src/diffusers/loaders/ip_adapter.py

```diff
@@ -228,27 +229,57 @@ def load_ip_adapter(
         unet = getattr(self, self.unet_name) if not hasattr(self, "unet") else self.unet
         unet._load_ip_adapter_weights(state_dicts, low_cpu_mem_usage=low_cpu_mem_usage)
 
-    def set_ip_adapter_scale(self, scale):
+    def set_ip_adapter_scale(self, scale: Union[float, Dict, List[Union[float, Dict]]], default_scale=0.0):
```

Comment:
Why use `default_scale`?

Reply:
`default_scale` came from this discussion:

https://github.com/huggingface/diffusers/pull/7668#discussion_r1566641088

I still have my doubts about this, but basically we already have a default of 1.0 for LoRAs but @DannHuang  wants  the default for `IP Adapters` to be 0.0.

IMO I like it both to be 0.0 because people that wan't to tinker with this, probably want to test each block and if they already know which blocks to scale, they'll probably set them all.

Reply:
I want to add a little more context here too:

If we want to set the style with default to `0.0`:

```python
style_scale = [
    {"up": {"block_0": [0.0, 1.0, 0.0]}},
]

composition_scale = [
    {"down": {"block_2": [0.0, 1.0]}},
]
```

but with default to `1.0`:

```python
style_scale = {
    "down": 0.0,
    "mid": 0.0,
    "up": {"block_0": [0.0, 1.0, 0.0], "block_1": 0.0},
}
composition_scale = {
    "down": {"block_1": 0.0, "block_2": [0.0, 1.0]},
    "mid": 0.0,
    "up": 0.0,
}
```

In terms of popularity, IP Adapters for style and composition is a lot more popular than block wise LoRAs, so it's going to be used quite a lot.

Reply:
But `default_scale` should not lead to breaking changes. This means `default_scale` should be 1.0 for LoRA stuff as that's what has been used by the community.

Reply:
I agree with @asomoza from the aspect setting scale for IP-Adapter/LoRAs. IMO in such scenario we are anticipating the effects from the layers we specified in the input config, so the rest of them should be 0. But indeed, making the ```default_scale``` default to 1 will not change the behavior of existing codes.